src/test/ui/feature-gates/feature-gate-const_transmute.rs

```diff
@@ -4,6 +4,8 @@ use std::mem;
 struct Foo(u32);
 
 const TRANSMUTED_U32: u32 = unsafe { mem::transmute(Foo(3)) };
-//~^ ERROR `std::intrinsics::transmute` is not yet stable as a const fn
+
+const fn transmute_fn() -> u32 { unsafe { mem::transmute(Foo(3)) } }
```

Comment:
Maybe also add a test without the `unsafe` block, and one that directly calls `core::intrinsics::transmute` (which is "accidentally" accessible on stable)? And... hm, what other combinations could we try?